src/lib389/lib389/cli_ctl/dblib.py

```diff
@@ -48,6 +62,47 @@ def __init__(self, *args, **kwargs):
         self.__dict__ = self
 
 
+def get_bdb_impl_status():
+    # Check installed rpms
+    try:
+        from rpm import TransactionSet
+        ts = TransactionSet()
+        pkgs = { pkgname:(len ([ pkg for pkg in ts.dbMatch( 'name', \
+                 pkgname ) ]) > 0) for pkgname in TESTED_PACKAGES}
+        ts.closeDB()
+    except ImportError:
+        if os.path.isfile(RPM_TOOL):
+            pkgs = { pkgname:( subprocess.run([RPM_TOOL, '-q', pkgname], \
+                     stdin=subprocess.DEVNULL,stderr=subprocess.DEVNULL \
+                     ).returncode  == 0 ) for pkgname in TESTED_PACKAGES }
+        else:
+            pkgs = { pkg:None for pkg in TESTED_PACKAGES}
+
+    # Check libback-ldbm plugin content
+    p = Paths()
+    plugin = None
+    BDB_IMPL_STATUS
+    for pdir in PLUGIN_PATHS:
+        for ppath in glob.glob(f'{pdir}/libback-ldbm.*'):
+            plugin = ppath
+    if plugin and os.path.isfile(OBJDUMP_TOOL):
+        rc = subprocess.run(['objdump', '-R', plugin], text=True, capture_output=True)
```

Comment:
Instead of testing for symbols, can we make a decision about type of libdb based on `/proc/$(pidof ns-slapd)/maps` contents?

For example:
* bundled libdb
```
grep libdb /proc/$(pidof ns-slapd)/maps
7fdd4d9eb000-7fdd4da0f000 r--p 00000000 fd:02 4623526                    /usr/lib64/dirsrv/libdb-5.3-389ds.so
7fdd4da0f000-7fdd4db61000 r-xp 00024000 fd:02 4623526                    /usr/lib64/dirsrv/libdb-5.3-389ds.so
7fdd4db61000-7fdd4dba7000 r--p 00176000 fd:02 4623526                    /usr/lib64/dirsrv/libdb-5.3-389ds.so
7fdd4dba7000-7fdd4dbb1000 r--p 001bb000 fd:02 4623526                    /usr/lib64/dirsrv/libdb-5.3-389ds.so
7fdd4dbb1000-7fdd4dbb2000 rw-p 001c5000 fd:02 4623526                    /usr/lib64/dirsrv/libdb-5.3-389ds.so
```

* system libdb
```
grep libdb /proc/$(pidof ns-slapd)/maps
ffffa5470000-ffffa549b000 r--p 00000000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa549b000-ffffa54a0000 ---p 0002b000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa54a0000-ffffa55d4000 r-xp 00030000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa55d4000-ffffa55e0000 ---p 00164000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa55e0000-ffffa562a000 r--p 00170000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa562a000-ffffa5636000 ---p 001ba000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa5636000-ffffa5640000 r--p 001c6000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
ffffa5640000-ffffa5641000 rw-p 001d0000 fd:00 1181522                    /usr/lib64/libdb-5.3.so
```

And with no libdb available (or if it's already an instance with mdb), there wouldn't be any output.
The only caveat is that the instance needs to be running...

Reply:
Interesting idea,. IMHO we can even keep the benefit of  both solutions:
  Keep current code and if status is UNKNOWN then try to  get the process maps 
  
  Another solution would be to use ldd on libback-ldbm.so since ldd is provided by glibc-common that should always be installed